src/nucleus/eval.ml

```diff
@@ -352,6 +352,12 @@ let rec infer env (c',loc) =
     let j = Judgement.mk_term ctx te ty in
     Value.return_term j
 
+  | Syntax.Yield ->
+    begin match Environment.lookup_continuation env with
+      | Some y -> Value.return y
+      | None -> Error.impossible ~loc "yield without continuation set"
```

Comment:
Shouldn't this be a runtime error? It's possible for user to use `yield` outside a handler.